src/storage-client/src/sink.rs

```diff
@@ -584,57 +606,86 @@ pub async fn determine_sink_resume_upper(
         // sink using the same progress topic crashes in that small window
         // between writing the progress record and committing the transaction,
         // in which case we have to wait out the transaction timeout.
-        //
-        // Important invariant: we only exit this loop successfully (i.e., not
-        // returning an error) if we have positive proof of a position at or
-        // beyond the high water mark. To make this invariant easy to check, do
-        // not use `break` in the body of the loop.
         let mut last_upper = None;
-        while get_position()? < hi {
-            let message = match progress_client_read_committed.poll(progress_record_fetch_timeout) {
-                Some(Ok(message)) => message,
-                Some(Err(KafkaError::PartitionEOF(_))) => {
-                    // No message, but the consumer's position may have advanced
-                    // past a transaction control message that positions us at
-                    // or beyond the high water mark. Go around the loop again
-                    // to check.
+        const CHUNK_SIZE: i64 = 2048;
```

Comment:
Any takes for what is a sensible value for this parameter?

Reply:
This seems like a good start!

can you make it a `dyncfg`? `mz_storage_types::dyncfgs` should make this way easier than the past

Reply:
Yeah agreed on making this a dyncfg! Then we don't need to think about it too deeply to start.